_posts/2016-09-29-Rust-1.12.md

```diff
@@ -93,34 +93,31 @@ upcoming changes to the compiler enabled by MIR is a rewrite of the pass that
 generates LLVM IR, what `rustc` calls "translation", and after many months of
 effort the MIR-based backend has proved itself ready for prime-time.
 
-TODO: Improve code complexity: size of old trans vs size of new trans.
-
 MIR exposes perfect information about the program's control flow, so the
 compiler knows exactly whether types are moved or not. This means that it knows
-statically whether or not the value's destructor needs to be run. In cases where
-a value may or may not be moved at the end of a scope, the compiler now simply
-uses a single bitflag on the stack, which is in turn easier for optimization
-passes in LLVM to reason about. The end result is less work for the compiler and
-less bloat at runtime.
-
-The improvements in compile times of some workloads due to this change alone are
-significant:
-
-|               | 1.11   | 1.12  | compile time  |
-|---------------|--------|-------|---------------|
-| inflate-0.1.0 | 10.17s | 4.32s | 2.35x faster  |
-| hyper-0.5.0   | 24.03s | 5.23s | 4.58x faster  |
-| helloworld     | 0.17s  | 0.14s | 1.20x faster |
-| regex 0.1.30  | 9.63s  | 2.39s | 4.03x faster  |
+statically whether or not the value's destructor needs to be run. In cases
+where a value may or may not be moved at the end of a scope, the compiler now
+simply uses a single bitflag on the stack, which is in turn easier for
+optimization passes in LLVM to reason about. The end result is less work for
+the compiler and less bloat at runtime. In addition, because MIR is a simpler
+'language' than the full AST, it's also easier to implement compiler passes on,
+and easier to verify that they are correct.
+
+The improvements in compile times of some workloads due to this change alone
+are significant:
+
+|               | 1.11   | 1.12  | compile time |
+|---------------|--------|-------|--------------|
+| inflate-0.1.0 | 10.17s | 4.32s | 2.35x faster |
+| hyper-0.5.0   | 24.03s | 5.23s | 4.58x faster |
```

Comment:
@jonathandturner The improvements you are seeing in nightly are likely due to https://github.com/rust-lang/rust/pull/36524